drivers/counter/counter_mcux_pit.c

```diff
@@ -0,0 +1,262 @@
+			return -ENOTSUP;
+		}
+		PIT_StopTimer(config->base, channel);
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+		PIT_StartTimer(config->base, channel);
+	} else {
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+	}
+
+	return 0;
+}
+
+static uint32_t mcux_pit_get_pending_int(struct device *dev)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	uint32_t mask = PIT_TFLG_TIF_MASK;
+	uint32_t flags;
+
+	flags = PIT_GetStatusFlags(config->base, config->pit_channel);
+
+	return ((flags & mask) == mask);
+}
+
+static uint32_t mcux_pit_get_max_relative_alarm(struct device *dev)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+
+	return config->info.max_top_value;
+}
+
+static void mcux_pit_isr(void *arg)
+{
+	struct device *dev = arg;
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+	uint32_t flags;
+	uint32_t current = 0;
+
+	LOG_DBG("pit counter isr");
+	flags = PIT_GetStatusFlags(config->base, config->pit_channel);
+	PIT_ClearStatusFlags(config->base, config->pit_channel, flags);
+	if (data->top_callback) {
+		data->top_callback(dev, data->top_user_data);
+	}
+	if (data->alarm_callback) {
+		PIT_StopTimer(config->base, config->pit_channel);
+		mcux_pit_get_value(dev, &current);
+		data->alarm_callback(dev, config->pit_channel, current,
+				     data->alarm_user_data);
+	}
+}
+
+static int mcux_pit_set_alarm(struct device *dev, uint8_t chan_id,
+			      const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+
+	uint32_t ticks = alarm_cfg->ticks;
+
+	if (chan_id != CONFIG_PIT_CHANNEL) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	if (ticks > mcux_pit_get_top_value(dev)) {
+		LOG_ERR("Invalid tciks");
+		return -EINVAL;
+	}
+
+	PIT_StopTimer(config->base, chan_id);
+	PIT_SetTimerPeriod(config->base, chan_id, ticks);
+	data->alarm_callback = alarm_cfg->callback;
+	data->alarm_user_data = alarm_cfg->user_data;
+	LOG_DBG("set alarm to %d", ticks);
+	PIT_StartTimer(config->base, chan_id);
+	return 0;
+}
+
+static int mcux_pit_cancel_alarm(struct device *dev, uint8_t chan_id)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+
+	if (chan_id != CONFIG_PIT_CHANNEL) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	PIT_DisableInterrupts(config->base, chan_id, kPIT_TimerInterruptEnable);
+	PIT_StopTimer(config->base, chan_id);
+	data->alarm_callback = NULL;
+
+	return 0;
+}
+
+static int mcux_pit_init(struct device *dev)
+{
+	struct mcux_pit_config *config =
+		(struct mcux_pit_config *)dev->config_info;
+	pit_config_t pit_config;
+
+	PIT_GetDefaultConfig(&pit_config);
+	pit_config.enableRunInDebug = config->enableRunInDebug;
+
+	PIT_Init(config->base, &pit_config);
+
+	config->irq_config_func(dev);
+
+	/*hard code to 1s*/
```

Comment:
Why? This should probably be a dts property in `dts/bindings/rtc/nxp,kinetis-pit.yaml`

Reply:
ok